pkg/actions/addon/addon.go

```diff
@@ -94,13 +95,26 @@ func (a *Manager) getLatestVersion(addon *api.Addon) (string, error) {
 		return "", fmt.Errorf("no versions available for %q", addon.Name)
 	}
 
+	addonVersion := addon.Version
 	var versions []*version.Version
 	for _, addonVersionInfo := range addonInfos.Addons[0].AddonVersions {
 		v, err := version.NewVersion(*addonVersionInfo.AddonVersion)
 		if err != nil {
 			return "", fmt.Errorf("failed to parse version %q: %w", *addonVersionInfo.AddonVersion, err)
 		}
-		versions = append(versions, v)
+
+		switch addonVersion {
+		case "latest":
+			versions = append(versions, v)
+		default:
+			if strings.Contains(*addonVersionInfo.AddonVersion, addonVersion) {
+				versions = append(versions, v)
+			}
+		}
+	}
+
```

Comment:
does this mean we allow any version value that is a substring of an actual version value? e.g `eks` would be a valid version value now, or just `1.` ?

Reply:
`eks` should not be a valid version value and I just realised that this edge case is not handled (I thought it was), thank you! if the given version is not `latest`, the input version should be parsed with something like `version.NewVersion(version)` to make sure that it is a valid version. 

just `1` would find the latest version that starts with that, yes!

Reply:
> `eks` should not be a valid version value and I just realised that this edge case is not handled (I thought it was), thank you! if the given version is not `latest`, the input version should be parsed with something like `version.NewVersion(version)` to make sure that it is a valid version.

:+1: 

> just `1` would be a valid version value, the latest version that starts with `1` would be returned! is that ok?

mhmm good question, I think the UX I've seen before is being able to set `latest` or atleast `major.minor`, so `1.2` would be valid, but `1` wouldn't be. However I don't think its really worth having too much of a fuss about it. If the `NewVersion` can parse the inputs, thats probably good enough. I would of *guessed* that `1` isn't a valid input to it, but `1.0` is but I could be wrong :smile:

Reply:
just added a small thing so that anything that can't be parsed returns an error `malformed version: <version>`! the versioning library is so handy! 😄 

just checked what happens when the value input is `1` and it doesn't return any error, it just discovers the latest `1.x` version. I don't mind the UX - I guess it would be unusual to ask for a major version `1` or `2`, it wouldn't necessarily be wrong to do that, right?

Reply:
yeah I think its OK, but if people actually do this I will judge them :joy: